### PR TITLE
Jenkins pipeline first step: run Terraform syntax checker

### PR DESCRIPTION
**Pull request by alexsomesan**
This change introduces a simple Jenkins Pipeline DSL file that right now just runs one test: the Terraform syntax check.
Although not much of a test in itself, this serves a good base for tweaking Jenkins to run PR checks and other tests safely.
Once this is in place, we I will start adding Tectonic tests.

### CODE REVIEW

**Comment by alexsomesan** [comment]
cc: @mxinden @sym3tri @Quentin-M @bison

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+pipeline {
+  agent {
+    docker {
+      image 'quay.io/coreos/tectonic-terraform'
```
Comment (Quentin-M) [inline Jenkinsfile]: Tag?

**Comment by Quentin-M** [comment]
It looks good as it seems very straight forward to integrate with the upcoming https://github.com/coreos-inc/tectonic/blob/master/Jenkinsfile (sometimes next week).

File: Jenkinsfile (hunk 1 above)
Comment (alexsomesan) [inline Jenkinsfile]: Done

**Pull request merged**